### PR TITLE
LOK-2562 Save metrics thresholds related data  while creating  monitoring policiy

### PR DESCRIPTION
**Pull request by jahanzebdataq**
## Description
While creating monitoring policy we have different options for creating monitoring policies rules.

like internal , snmp trap , syslog and metrics thresholds.

when user select metrics threshold option then alert condition section appears. There are fields like enable, condition and thresholds that need to be saved while creating monitoring policy.

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2562

### CODE REVIEW

File: alert/src/main/java/org/opennms/horizon/alertservice/db/entity/ThresholdMetric.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the
+ * License.
+ */
+package org.opennms.horizon.alertservice.db.entity;
+
+import jakarta.persistence.*;
```
Comment (smunir-onms) [inline alert/src/main/java/org/opennms/horizon/alertservice/db/entity/ThresholdMetric.java:24]: please don't use wildcard imports, use individual class imports

File: alert/src/main/java/org/opennms/horizon/alertservice/mapper/ThresholdMetricMapper.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the
+ * License.
+ */
+package org.opennms.horizon.alertservice.mapper;
+
+import org.mapstruct.*;
```
Comment (smunir-onms) [inline alert/src/main/java/org/opennms/horizon/alertservice/mapper/ThresholdMetricMapper.java:24]: use individual class imports

**Review by smunir-onms — COMMENTED** [review]
LGTM, added comments for very minor things

**Review by aurang-zeb313 — COMMENTED** [review]
LGTM, Please add testing screen shots.

File: alert/src/main/resources/db/changelog/hs-0.1.0/tables/alert_condition_event_definition.xml [hunk 3]
```diff
@@ -26,7 +26,6 @@
             WHERE replace(ac.clear_event_type, '_', ' ') = ed.name;
         </sql>
 
-        <addNotNullConstraint tableName="alert_condition" columnName="trigger_event_definition_id"/>
```
[the remark this answers is not in the document]
Comment (jahanzebdataq) [reply alert/src/main/resources/db/changelog/hs-0.1.0/tables/alert_condition_event_definition.xml]: Done

File: alert/src/main/java/org/opennms/horizon/alertservice/mapper/ThresholdMetricMapper.java (hunk 2 above)
Comment (jahanzebdataq) [reply alert/src/main/java/org/opennms/horizon/alertservice/mapper/ThresholdMetricMapper.java]: Done

File: alert/src/main/java/org/opennms/horizon/alertservice/db/repository/ThresholdMatricRepository.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ * License.
+ */
+package org.opennms.horizon.alertservice.db.repository;
+
+import org.opennms.horizon.alertservice.db.entity.ThresholdMetric;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface ThresholdMatricRepository extends JpaRepository<ThresholdMetric, Long> {}
```
Comment (synqotik) [inline alert/src/main/java/org/opennms/horizon/alertservice/db/repository/ThresholdMatricRepository.java:29]: Fix name to `ThresholdMetricRepository`

Comment (jahanzebdataq) [reply alert/src/main/java/org/opennms/horizon/alertservice/db/repository/ThresholdMatricRepository.java:29]: Done

**Comment by jahanzebdataq** [comment]
> LGTM, Please add testing screen shots.

Screenshot from 2024-06-12 15-11-52
Screenshot from 2024-06-12 15-10-45
Screenshot from 2024-06-12 15-10-18
Screenshot from 2024-06-12 15-09-06
Screenshot from 2024-06-12 15-08-57
Screenshot from 2024-06-12 15-08-41
Screenshot from 2024-06-12 15-08-15

**Pull request merged**